sim-lib/src/sim_node.rs

```diff
@@ -1035,3 +1038,742 @@ impl UtxoLookup for UtxoValidator {
+
+        // Create channels in a ring so that we'll get long payment paths.
+        let mut node_1 = first_node;
+        for i in 0..n {
+            // Generate a new random node pubkey.
+            let (_, node_2) = get_random_keypair();
+
+            let node_1_to_2 = ChannelPolicy {
+                pubkey: node_1,
+                max_htlc_count: 483,
+                max_in_flight_msat: capacity_msat / 2,
+                min_htlc_size_msat: 1,
+                max_htlc_size_msat: capacity_msat / 2,
+                cltv_expiry_delta: 40,
+                base_fee: 1000 * i,
+                fee_rate_prop: 1500 * i,
+            };
+
+            let node_2_to_1 = ChannelPolicy {
+                pubkey: node_2,
+                max_htlc_count: 483,
+                max_in_flight_msat: capacity_msat / 2,
+                min_htlc_size_msat: 1,
+                max_htlc_size_msat: capacity_msat / 2,
+                cltv_expiry_delta: 40 + 10 * i as u32,
+                base_fee: 2000 * i,
+                fee_rate_prop: i,
+            };
+
+            channels.push(SimulatedChannel {
+                capacity_msat,
+                // Unique channel ID per link.
+                short_channel_id: ShortChannelID::from(i),
+                node_1: ChannelState::new(node_1_to_2, capacity_msat),
+                node_2: ChannelState::new(node_2_to_1, 0),
+            });
+
+            // Progress source ID to create a chain of nodes.
+            node_1 = node_2;
+        }
+
+        channels
+    }
+
+    macro_rules! assert_channel_balances {
+        ($channel_state:expr, $local_balance:expr, $in_flight_len:expr, $in_flight_total:expr) => {
+            assert_eq!($channel_state.local_balance_msat, $local_balance);
+            assert_eq!($channel_state.in_flight.len(), $in_flight_len);
+            assert_eq!($channel_state.in_flight_total(), $in_flight_total);
+        };
+    }
+
+    /// Tests state updates related to adding and removing HTLCs to a channel.
+    #[test]
+    fn test_channel_state_transitions() {
+        let local_balance = 100_000_000;
+        let mut channel_state =
+            ChannelState::new(create_test_policy(local_balance / 2), local_balance);
+
+        // Basic sanity check that we Initialize the channel correctly.
+        assert_channel_balances!(channel_state, local_balance, 0, 0);
+
+        // Add a few HTLCs to our internal state and assert that balances are as expected. We'll test
+        // `check_outgoing_addition` in more detail in another test, so we just assert that we can add the htlc in
+        // this test.
+        let hash_1 = PaymentHash { 0: [1; 32] };
+        let htlc_1 = Htlc {
+            amount_msat: 1000,
+            cltv_expiry: 40,
+        };
+
+        assert!(channel_state.check_outgoing_addition(&htlc_1).is_ok());
+        assert!(channel_state.add_outgoing_htlc(hash_1, htlc_1).is_ok());
+        assert_channel_balances!(
```

Comment:
Why the call to `check_outgoing_addition(...)` if the method is called in `add_outgoing_htlc(...)`? Wouldn't the result of the assertion be the same if a single (the second) call is made?

Reply:
Indeed, don't need both calls!